app/src/main/java/io/homeassistant/companion/android/notifications/NotificationActionReceiver.kt

```diff
@@ -42,32 +46,46 @@ class NotificationActionReceiver : BroadcastReceiver() {
             return
         }
 
-        when (intent.action) {
-            FIRE_EVENT -> fireEvent(notificationAction)
-            OPEN_URI -> openUri(context, notificationAction)
-        }
-
         val messageId = intent.getIntExtra(EXTRA_NOTIFICATION_ID, -1)
-        if (messageId != -1) {
-            val notificationService: NotificationManager =
-                context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
-            notificationService.cancel(messageId)
+        val onComplete: () -> Unit = {
+            (context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager)
+                .cancel(messageId)
+        }
+        val onFailure: () -> Unit = {
+            Handler(context.mainLooper).post {
+                Toast.makeText(context, R.string.event_error, Toast.LENGTH_LONG).show()
+            }
+        }
+        when (intent.action) {
+            FIRE_EVENT -> fireEvent(notificationAction, onComplete, onFailure)
+            OPEN_URI -> openUri(context, notificationAction, onComplete)
         }
     }
 
-    private fun fireEvent(action: NotificationAction) {
+    private fun fireEvent(
+        action: NotificationAction,
+        onComplete: () -> Unit,
+        onFailure: () -> Unit
+    ) {
         ioScope.launch {
-            integrationUseCase.fireEvent(
-                "mobile_app_notification_action",
-                action.data.plus(Pair("action", action.key))
-            )
+            try {
+                integrationUseCase.fireEvent(
+                    "mobile_app_notification_action",
+                    action.data.plus(Pair("action", action.key))
+                )
+                onComplete()
+            } catch (e: Exception) {
+                Log.e(TAG, "Unable to fire event.", e)
```

Comment:
Should this be the same `event_error` string ?

Reply:
Since this is just the ADB log it should never get shown to an end user.